src/main/java/org/wildfly/security/sasl/gssapi/GssapiServer.java

```diff
@@ -58,19 +63,34 @@
 
     GssapiServer(final String protocol, final String serverName, final Map<String, ?> props,
             final CallbackHandler callbackHandler) throws SaslException {
-        super(SaslMechanismInformation.Names.GSSAPI, protocol, serverName, props, callbackHandler, log);
+        super(GSSAPI, protocol, serverName, props, callbackHandler, log);
 
         // Initialise our GSSContext
         GSSManager manager = GSSManager.getInstance();
 
-        // According to the Javadoc we will have a protocol and server name.
-        String localName = protocol + "@" + serverName;
-        log.tracef("Our name '%s'", localName);
         GSSContext gssContext = null;
+
+        GSSCredential ourCredential = null;
+
+        ServerCredentialCallback gssCredentialCallback = new ServerCredentialCallback(GSSKerberosCredential.class);
+        try {
+            log.trace("Obtaining GSSCredential for the service from callback handler...");
+            callbackHandler.handle(new Callback[] { gssCredentialCallback });
+            ourCredential = gssCredentialCallback.applyToCredential(GSSKerberosCredential.class, GSSKerberosCredential::getGssCredential);
+        } catch (IOException | UnsupportedCallbackException e) {
+            log.trace("Unable to obtain GSSCredential from CallbackHandler", e);
```

Comment:
I think if IOException is thrown then the authentication should fail, because the callback handler is signalling a problem as opposed to not supporting the operation.  WDYT?